### ISSUE TITLE
Undefined metric value, caused by empty GTs or predictions After validation

### ISSUE THREAD

**Issue by Body123** [mindee/doctr#1975]
### Bug description

After validation completed, the following error occurred: Epoch 5/5 - Validation loss: 1.00117 (Undefined metric value, caused by empty GTs or predictions also I need to ask about the logs and plot of training where I can find it.
------------------------------------------------------------------------------------
[DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00188:  62%|█████████████████████████████████████████████████████████▌                                   | 13/21 [00:05<00:01,  6.58it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00127:  62%|█████████████████████████████████████████████████████████▌                                   | 13/21 [00:05<00:01,  6.58it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00087:  71%|██████████████████████████████████████████████████████████████████▍                          | 15/21 [00:05<00:00,  8.39it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00154:  71%|██████████████████████████████████████████████████████████████████▍                          | 15/21 [00:05<00:00,  8.39it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00201:  81%|███████████████████████████████████████████████████████████████████████████▎                 | 17/21 [00:05<00:00, 10.25it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00084:  81%|███████████████████████████████████████████████████████████████████████████▎                 | 17/21 [00:05<00:00, 10.25it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00078:  81%|███████████████████████████████████████████████████████████████████████████▎                 | 17/21 [00:05<00:00, 10.25it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00201:  95%|████████████████████████████████████████████████████████████████████████████████████████▌    | 20/21 [00:06<00:00, 12.77it/s][DEBUG] Predictions count: 1 for current batch
Validation loss: 1.00079: 100%|█████████████████████████████████████████████████████████████████████████████████████████████| 21/21 [00:06<00:00,  3.44it/s]
Epoch 5/5 - Validation loss: 1.00117 (Undefined metric value, caused by empty GTs or predictions)

### Code snippet to reproduce the bug

python references/detection/train_pytorch.py db_resnet50 \
  --train_path my_dataset_for_doctr/data_split/train \
  --val_path my_dataset_for_doctr/data_split/val \
  --epochs 5

### Error traceback

--

### Environment

After validation completed, the following error occurred: Epoch 5/5 - Validation loss: 1.00117 (Undefined metric value, caused by empty GTs or predictions also I need to ask about the logs and plot of training where I can find it.
------------------------------------------------------------------------------------
[DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00188:  62%|█████████████████████████████████████████████████████████▌                                   | 13/21 [00:05<00:01,  6.58it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00127:  62%|█████████████████████████████████████████████████████████▌                                   | 13/21 [00:05<00:01,  6.58it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00087:  71%|██████████████████████████████████████████████████████████████████▍                          | 15/21 [00:05<00:00,  8.39it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00154:  71%|██████████████████████████████████████████████████████████████████▍                          | 15/21 [00:05<00:00,  8.39it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00201:  81%|███████████████████████████████████████████████████████████████████████████▎                 | 17/21 [00:05<00:00, 10.25it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00084:  81%|███████████████████████████████████████████████████████████████████████████▎                 | 17/21 [00:05<00:00, 10.25it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00078:  81%|███████████████████████████████████████████████████████████████████████████▎                 | 17/21 [00:05<00:00, 10.25it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00201:  95%|████████████████████████████████████████████████████████████████████████████████████████▌    | 20/21 [00:06<00:00, 12.77it/s][DEBUG] Predictions count: 1 for current batch
Validation loss: 1.00079: 100%|█████████████████████████████████████████████████████████████████████████████████████████████| 21/21 [00:06<00:00,  3.44it/s]
Epoch 5/5 - Validation loss: 1.00117 (Undefined metric value, caused by empty GTs or predictions)

### Deep Learning backend

After validation completed, the following error occurred: Epoch 5/5 - Validation loss: 1.00117 (Undefined metric value, caused by empty GTs or predictions also I need to ask about the logs and plot of training where I can find it.
------------------------------------------------------------------------------------
[DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00188:  62%|█████████████████████████████████████████████████████████▌                                   | 13/21 [00:05<00:01,  6.58it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00127:  62%|█████████████████████████████████████████████████████████▌                                   | 13/21 [00:05<00:01,  6.58it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00087:  71%|██████████████████████████████████████████████████████████████████▍                          | 15/21 [00:05<00:00,  8.39it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00154:  71%|██████████████████████████████████████████████████████████████████▍                          | 15/21 [00:05<00:00,  8.39it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00201:  81%|███████████████████████████████████████████████████████████████████████████▎                 | 17/21 [00:05<00:00, 10.25it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00084:  81%|███████████████████████████████████████████████████████████████████████████▎                 | 17/21 [00:05<00:00, 10.25it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00078:  81%|███████████████████████████████████████████████████████████████████████████▎                 | 17/21 [00:05<00:00, 10.25it/s][DEBUG] Predictions count: 2 for current batch
Validation loss: 1.00201:  95%|████████████████████████████████████████████████████████████████████████████████████████▌    | 20/21 [00:06<00:00, 12.77it/s][DEBUG] Predictions count: 1 for current batch
Validation loss: 1.00079: 100%|█████████████████████████████████████████████████████████████████████████████████████████████| 21/21 [00:06<00:00,  3.44it/s]
Epoch 5/5 - Validation loss: 1.00117 (Undefined metric value, caused by empty GTs or predictions)

**Comment by felixdittrich92** [comment]
You have two options to log

`--wb` to log to weights&biases

or

`--clearml` to log to ClearML


I will close the issue because it's duplicate of #1974
